js/plugins/googleai/src/index.ts

```diff
@@ -81,82 +82,127 @@ export interface PluginOptions {
-      Object.keys(EMBEDDER_MODELS).forEach((name) =>
-        defineGoogleAIEmbedder(ai, name, { apiKey: options?.apiKey })
-      );
-    }
+  if (apiVersions.includes('v1beta')) {
+    Object.keys(SUPPORTED_V15_MODELS).forEach((name) =>
+      defineGoogleAIModel({
+        ai,
+        name,
+        apiKey: options?.apiKey,
+        apiVersion: 'v1beta',
+        baseUrl: options?.baseUrl,
+        debugTraces: options?.experimental_debugTraces,
+      })
+    );
+  }
+  if (apiVersions.includes('v1')) {
+    Object.keys(SUPPORTED_V1_MODELS).forEach((name) =>
+      defineGoogleAIModel({
+        ai,
+        name,
+        apiKey: options?.apiKey,
+        apiVersion: undefined,
+        baseUrl: options?.baseUrl,
+        debugTraces: options?.experimental_debugTraces,
+      })
+    );
+    Object.keys(SUPPORTED_V15_MODELS).forEach((name) =>
+      defineGoogleAIModel({
+        ai,
+        name,
+        apiKey: options?.apiKey,
+        apiVersion: undefined,
+        baseUrl: options?.baseUrl,
+        debugTraces: options?.experimental_debugTraces,
+      })
+    );
+    Object.keys(EMBEDDER_MODELS).forEach((name) =>
+      defineGoogleAIEmbedder(ai, name, { apiKey: options?.apiKey })
+    );
+  }
 
-    if (options?.models) {
-      for (const modelOrRef of options?.models) {
-        const modelName =
-          typeof modelOrRef === 'string'
-            ? modelOrRef
-            : // strip out the `googleai/` prefix
-              modelOrRef.name.split('/')[1];
-        const modelRef =
-          typeof modelOrRef === 'string' ? gemini(modelOrRef) : modelOrRef;
-        defineGoogleAIModel({
-          ai,
-          name: modelName,
-          apiKey: options?.apiKey,
-          baseUrl: options?.baseUrl,
-          info: {
-            ...modelRef.info,
-            label: `Google AI - ${modelName}`,
-          },
-          debugTraces: options?.experimental_debugTraces,
-        });
-      }
+  if (options?.models) {
+    for (const modelOrRef of options?.models) {
+      const modelName =
+        typeof modelOrRef === 'string'
+          ? modelOrRef
+          : // strip out the `googleai/` prefix
+            modelOrRef.name.split('/')[1];
+      const modelRef =
+        typeof modelOrRef === 'string' ? gemini(modelOrRef) : modelOrRef;
+      defineGoogleAIModel({
+        ai,
+        name: modelName,
+        apiKey: options?.apiKey,
+        baseUrl: options?.baseUrl,
+        info: {
+          ...modelRef.info,
+          label: `Google AI - ${modelName}`,
+        },
+        debugTraces: options?.experimental_debugTraces,
+      });
     }
-  });
+  }
+}
+
+async function resolver(
+  ai: Genkit,
+  actionType: ActionType,
+  actionName: string,
+  options?: PluginOptions
+) {
+  // TODO: also support other actions like 'embedder'
+  switch (actionType) {
+    case 'model':
+      await resolveModel(ai, actionName, options);
+      break;
+    default:
+    // no-op
+  }
+}
+
+async function resolveModel(
```

Comment:
doesn't need to be async?

Reply:
doesn't need to be async... yet. Rather have it and not need it than the other way around.